### PR TITLE
Upgrade kserve python dep

### LINKED ISSUE #2041: Cannot install required version of numpy on M1 mac
/kind bug

Issue:
Installation on python 3.8 or 3.9 (and presumably all versions of Python) of the v0.8.0 release candidate fails due to the pinned requirement of numpy.

Expected behavior:
kserve's release candidate for 0.8 can be installed on an M1 mac.

Extra information:
https://github.com/numpy/numpy/releases/tag/v1.21.0 numpy 1.21+ allows installation on M1 macs.

**Environment:**

- OS (e.g. from `/etc/os-release`): M1 mac

### PR DESCRIPTION
**Pull request by Suresh-Nakkeran**
This PR is to upgrade KServe Python SDk dependencies.

fixes #2041

### CODE REVIEW

File: python/kserve/requirements.txt [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
-avro>=1.10.1
-boto3==1.20.24
-psutil>=5.0
+avro>=1.11.0
+boto3==1.21.0
+psutil>=5.9.0
 ray[serve]==1.9.2
 grpcio>=1.34.0
-google_api_core==1.29.0
+google_api_core==2.5.0
```
Comment (yuzisun) [inline python/kserve/requirements.txt]: we can remove pinned `google_api_core` as it is depended by `google-cloud-storage`.
https://github.com/googleapis/python-storage/blob/main/setup.py#L32

File: python/kserve/requirements.txt [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-google_auth==1.34.0
-cffi==1.14.6
-cryptography==3.4.7
-idna==3.2
-certifi==2021.5.30
-azure_core==1.17.0
-tritonclient==2.14.2
+googleapis_common_protos==1.54.0
+cachetools==5.0.0
+google_auth==2.6.0
```
Comment (yuzisun) [inline python/kserve/requirements.txt]: we can remove pinned `google_auth` as it is depended by `google-cloud-storage`.
https://github.com/googleapis/python-storage/blob/main/setup.py#L31

File: python/kserve/requirements.txt [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
+cffi==1.15.0
+cryptography==36.0.1
```
Comment (yuzisun) [inline python/kserve/requirements.txt]: We can remove pinned `cryptography` as it is depended by `azure-storage-blob`
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/setup.py#L81

File: python/kserve/requirements.txt [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
+idna==3.3
+certifi==2021.10.8
+azure_core==1.23.0
```
Comment (yuzisun) [inline python/kserve/requirements.txt]: We can remove pinned `azure_core` as it is depended by `azure-storage-blob`
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/setup.py#L79

Comment (Suresh-Nakkeran) [reply python/kserve/requirements.txt]: removed!

File: python/kserve/requirements.txt (hunk 3 above)
Comment (Suresh-Nakkeran) [reply python/kserve/requirements.txt]: removed!

File: python/kserve/requirements.txt (hunk 2 above)
Comment (Suresh-Nakkeran) [reply python/kserve/requirements.txt]: removed!

File: python/kserve/requirements.txt (hunk 1 above)
Comment (Suresh-Nakkeran) [reply python/kserve/requirements.txt]: removed!

**Pull request merged**